src/cargo/util/rustc.rs

```diff
@@ -82,17 +82,19 @@ impl Rustc {
                 verbose_version
             )
         })?;
-        let commit_hash = extract("commit-hash: ").ok().map(|hash| {
-            debug_assert!(
-                hash.chars().all(|ch| ch.is_ascii_hexdigit()),
-                "commit hash must be a hex string, got: {hash:?}"
-            );
-            debug_assert!(
-                hash.len() == 40 || hash.len() == 64,
-                "hex string must be generated from sha1 or sha256 (i.e., it must be 40 or 64 characters long)\ngot: {hash:?}"
-            );
-            hash.to_string()
-        });
+        let commit_hash = extract("commit-hash: ").ok()
+            .filter(|hash| {
+                if !hash.chars().all(|ch| ch.is_ascii_hexdigit()) {
```

Comment:
Maybe add a comment explaining why the code does what it does now?

Reply:
Thanks. Updated!